### PR TITLE
Optimise query get pool distr

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

Improve the performance of `GetPoolDistr` by implementing it with `calculatePoolDistr'` instead of `calculatePoolDistr`.

No interface changes.

# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Block.hs [hunk 1]
```diff
@@ -99,6 +103,8 @@ class
     -- Hard-fork related constraints
   , HasPartialConsensusConfig proto
   , FromCBOR (SL.PState (ProtoCrypto proto))
+  , Share (Core.TxOut era) ~ Interns (SL.Credential 'SL.Staking (ProtoCrypto proto))
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Block.hs:106]: Where do these requirements come from? Seems unrelated to the rest of the diff.

Please continue the existing pattern by adding a comment that summarizes why they're needed and separates them from the "Hard-fork related constraints" above them.

**Review by nfrisby — DISMISSED** [review]
Just the one question regarding the constraints added to `ShelleyCompatible`.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Block.hs (hunk 1 above)
Comment (newhoggy) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Block.hs]: You're right.  I don't need them anymore.  Removed.

**Review by nfrisby — APPROVED** [review]
Approved. Thanks!

That's a scary Hydra failure message 😬 

```
/nix/store/6dpldw6wq7dh8zgp7x4q20xpiqlsp6aj-stdenv-linux/setup: line 1393: /nix/store/g7wiyprvpvgv2nsy70csfq1g080yhzdm-ouroboros-consensus-cardano-test-test-test-0.1.0.0/bin/test: cannot execute binary file: Exec format error
```
